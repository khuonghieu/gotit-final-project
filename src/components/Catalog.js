import React from 'react';
import { connect } from 'react-redux';
import { Header } from '@gotitinc/design-system';
import UserProfile from './User/UserProfile';
import UserSignOut from './User/UserSignOut';
import CategoriesTabList from './Categories/CategoriesTabList';

function mapStateToProps(state) {
  return {
    user: state.user,
  };
}

export function Catalog({ user }) {
  return (
    <div>
      <Header fullWidth>
        <Header.Main>
          <Header.Left>
            <UserProfile user={user} />
          </Header.Left>
          <Header.Right>
            <div><UserSignOut /></div>
          </Header.Right>
        </Header.Main>
      </Header>
      <hr />
      <div><CategoriesTabList /></div>
    </div>
  );
}

export default connect(mapStateToProps, null)(Catalog);
