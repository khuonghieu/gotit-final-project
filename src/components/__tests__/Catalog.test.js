import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Button, Header } from '@gotitinc/design-system';
import { Catalog } from '../Catalog';

configure({ adapter: new Adapter() });

describe('components/Catalog.js', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      user: {
        loggedIn: false,
        currentUser: null,
      },
      chooseModal: jest.fn(),
    };
  });

  const setup = () => {
    wrapper = shallow(
      <Catalog {...props} />,
    );
  };

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should show sign out button when logged in', () => {
    props = {
      ...props,
      user: {
        loggedIn: true,
        currentUser: null,
      },
    };
    setup();
    const signOutButton = wrapper.find(Header.Right);
    expect(signOutButton.length).toBe(1);
  });

  it('should show sign up and sign in button when not logged in', () => {
    props = {
      ...props,
      user: {
        loggedIn: false,
        currentUser: null,
      },
    };
    setup();
    const buttons = wrapper.find(Button);
    expect(buttons.length).toBe(2);
  });
});
