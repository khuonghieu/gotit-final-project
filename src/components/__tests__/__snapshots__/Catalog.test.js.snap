// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Catalog.js should render correctly 1`] = `ShallowWrapper {}`;
